Source/IntegrationTests/TestFiles/LitTests/LitTest/git-issues/git-issue-5017b.dfy.expect

```diff
@@ -0,0 +1,10 @@
+git-issue-5017b.dfy(31,14): Error: fully defined predicate 'Valid' is inherited from trait 'B' and is not allowed to be re-declared
+git-issue-5017b.dfy(34,21): Error: fully defined predicate 'Valid' is inherited from trait 'B' and is not allowed to be re-declared
+git-issue-5017b.dfy(34,21): Error: static lemma 'reveal_Valid' is inherited from trait 'B' and is not allowed to be re-declared
+git-issue-5017b.dfy(38,14): Error: overridden predicate 'Valid' in 'M0' must be 'opaque' since the member is 'opaque' in trait 'C'
+git-issue-5017b.dfy(44,14): Error: overridden predicate 'Valid' in 'M2' must be 'opaque' since the member is 'opaque' in trait 'C'
+git-issue-5017b.dfy(66,14): Error: overridden predicate 'Valid' in 'Y0' must be 'opaque' since the member is 'opaque' in trait 'C'
+git-issue-5017b.dfy(69,14): Error: overridden predicate 'Valid' in 'Y1' must be 'opaque' since the member is 'opaque' in trait 'C'
+git-issue-5017b.dfy(94,13): Error: member 'reveal_Valid' does not exist in trait 'A'
```

Comment:
I worried a bit about this issue. It seems that, to resolver it, one would need to implement this member, but the resolution is to actually. make `Valid` opaque in A. Could you detect that particular case and emit a more useful error message?

Reply:
Excellent idea. I improved many messages. However, it became more difficult for some of them. The problem is that, early on during resolution, we change `reveal X();` expressions into `reveal_X();` lemma calls. By the time that these attempted lemma calls go through the resolver, the name looked up is `reveal_X`. To improve the error message, we then have to detect that we're in a reveal-statement context and that the thing we're trying to look up has a name that starts with "`reveal_`" (which is not actually guaranteed to be one of the rewritten calls, since this could have been a user-defined name in the first place), and then remove the "`reveal_`" prefix, try to find what the name `X` might have referred to, and finally reconstruct the reason why no `reveal_X` lemma was created in the first place. It would have been much better not to have done the `reveal_X`-lemma surgery so early in the pipeline. Then, the symbol we'd be looking at during resolution would be `X`. (Btw, I've seen this mistake many times--a change is made too early in the resolution pipeline, because it _seems_ as if that's a quick way to get a new feature into the language, but then this bites later on.)

Anyhow, I feel okay with the error messages I did improve. At some point in the future, I expect that we'll revisit `reveal` statements to expand their role. That would be a good time to handle `reveal` in a way that's less hacky than today's `reveal_`-lemmas.